### PR TITLE
Various Improvements

### PR DESCRIPTION
**Pull request by AlicanC**
[feat: add craco and banner](https://github.com/FuelLabs/block-explorer-v2/commit/315e4defe70462f759e295998f9798ec5cbac272)

- Adds [CRACO](https://github.com/gsoft-inc/craco) so we can further tweak our builds
- Adds an HTML banner to build output

[chore: add VSCode config](https://github.com/FuelLabs/block-explorer-v2/commit/4a82a39ff8105e9afcb79afaa0febadbe491de1c)

- Some config from fuels-ts

[chore: improve ESLint and Prettier config](https://github.com/FuelLabs/block-explorer-v2/commit/f3fe5df2d6c2b55c784a9b3ed230e6265896fc51)

- More config from fuels-ts

[feat: lazy load Page components](https://github.com/FuelLabs/block-explorer-v2/commit/d379310b0e20898b8b07624d9aadfb7fe14ce6da)

- Makes Page components lazy load so instead of loading code for all pages, we load only for the page we are on

[chore: upgrade react-router-dom to v6.2.2](https://github.com/FuelLabs/block-explorer-v2/commit/8563d4bfe507ba9036aaf0ecfa2bc6c306d61655)

- Upgrades RR from 5 to 6

### CODE REVIEW

File: craco.config.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+// https://patorjk.com/software/taag/#p=display&f=ANSI%20Shadow&t=Fuel%20Labs
+const banner = `
+███████╗██╗   ██╗███████╗██╗         ██╗      █████╗ ██████╗ ███████╗
+██╔════╝██║   ██║██╔════╝██║         ██║     ██╔══██╗██╔══██╗██╔════╝
+█████╗  ██║   ██║█████╗  ██║         ██║     ███████║██████╔╝███████╗
+██╔══╝  ██║   ██║██╔══╝  ██║         ██║     ██╔══██║██╔══██╗╚════██║
+██║     ╚██████╔╝███████╗███████╗    ███████╗██║  ██║██████╔╝███████║
+╚═╝      ╚═════╝ ╚══════╝╚══════╝    ╚══════╝╚═╝  ╚═╝╚═════╝ ╚══════╝
+${packageJson.name} v${packageJson.version}
```
Comment (luizstacio) [inline craco.config.js:36]: 👌

File: tsconfig.json [hunk 2]
```diff
@@ -1,20 +1,4 @@
...
-    "forceConsistentCasingInFileNames": true,
-    "noFallthroughCasesInSwitch": true,
-    "module": "esnext",
-    "moduleResolution": "node",
-    "resolveJsonModule": true,
-    "isolatedModules": true,
-    "noEmit": true,
-    "jsx": "react-jsx"
-  },
+  "extends": "./tsconfig.base.json",
```
Comment (luizstacio) [inline tsconfig.json:2]: @AlicanC on this project as we don't have packages and stuff would be nice to keep tsconfig.json and tsconfig.base.json in a single file. But no big deal!

**Review by luizstacio — APPROVED** [review]
Nice improvements, just left a comment!

File: tsconfig.json (hunk 2 above)
Comment (AlicanC) [reply tsconfig.json:2]: We don't but we still need multiple configurations: one for files under `src`, which use `tsconfig.json` and another for the rest, for example for files under `script` or in repo root, which use `tsconfig.eslint.json`. Both of these configurations import `tsconfig.base.json`.

**Pull request merged**